### DISCUSSION TITLE
TPU training and WandB implementation causes "AssertionError: can only test a child process"

### DISCUSSION THREAD

**Discussion by adamDhalla** [Lightning-AI/pytorch-lightning#6639] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi all, 

I'm not raising this issue as a bug since it's already been raised [here,](https://forums.pytorchlightning.ai/t/tpu-failures-in-colab/310) and was seemingly solved. 

I have pretty much the exact problem - I'm trying to implement weights and biases with a TPU and it keeps showing "AssertionError: can only test a child process". The person who raised the old error also replicated the issue in the boring model [here.](https://colab.research.google.com/drive/1Bg-vw8eubL1Y8ksV0ZCLh3sQDx6awskK?usp=sharing)

The thing is he never explains how he solves the issue - he explains the issue itself, but doesn't say how he solves it. I'm wondering _where_ I should be initializing the wandb object: 
```
    config = {'hparam1' : number, 'hparam2' : number2 ..... }
    wandb_logger = WandbLogger(name='EXP_NAME',project='PROJECT_NAME',log_model=True,config=config)
```

Right now, I'm initializing both of those lines above the trainer. The trainer requires _wandb_logger_ be passed through, so I can't do the wandb_logger = WandbLogger... line any higher up. 

The whole problem seems to come about as a problem with distributed processing and WandB.

Thanks,
Adam

**Comment by morganmcg1** [comment]
@adamDhalla is this still an issue for you? I think wandb support for TPUs should be better in the latest version
